### ISSUE TITLE
Problems with micro-stuttering

### ISSUE THREAD

**Issue by jtrees** [elementary/os#355]
## Prerequisites
- [x] I have searched open and closed issues for duplicates.

## Describe the bug
Sometimes the desktop environment doesn't feel smooth. It's a bit hard to describe but it's like some things are capped to 30Hz or respond with a delay of a single frame. It's not consistent but feels like about 90% of the time that this micro-stuttering is present.

### To Reproduce
Steps to reproduce the behavior (any of the following)
- Open the applications menu
- Have the dock hide itself and/or return back to view
- Move a window

### Expected behavior
Everything* should respond immediately and animate smoothly (at the refresh rate of the monitor - probably 60 Hz for most users).

## Platform Information
![Screenshot from 2020-05-18 12-38-25](https://user-images.githubusercontent.com/5802758/82204088-8e078d80-9904-11ea-80b5-ba4b68261eec.png)

- [ ] I'm using the latest version from git that I've manually compiled
- [x] I'm using the latest released stable version

Using Nvidia propitiatory drivers, currently at version 440.82, but have used several different versions in the past.

## Additional context

- This behavior has been present for a long time. Possibly since Juno. It was not present on Loki.
- I use a dual screen setup.
- I believe many other users are experiencing similar issues so I'm trying to gather more information here: https://ethercalc.org/usfdfp40bdea

**Comment by KarkanAlzwayed** [comment]
I have the same experience. Maximizing windows and minimizing them cuts half way into the process. It stops half say then does the job. It is most noticeable when I use overview to show all the running apps. I thought it was my 4k screen, but it's running at 60. But now I think it is only on nvidia devices. I have another laptop without nvidia and it is smooth as ever. To eliminate it, I'd have to switch the screen resolution to 1150, make the refresh rate 120 and enable "LoDpi", but then the screen looks grainy. I have tried switching to the intel graphics, but that didn't help much. Some of the stutter goes away, but then the screen would look like it is only 30 FPS at all times.

**Comment by TheGag96** [comment]
I have a 1660 Ti, hooked up to two 60hz non-g/free-sync monitors, and am using the nvidia proprietary drivers also. Performance can be pretty choppy, and screen tearing in games is basically unavoidable - I really think the drivers are the culprit. I'm not sure what they can do to fix it.

**Comment by jtrees** [comment]
> I really think the drivers are the culprit. I'm not sure what they can do to fix it.

I don't know either. But doing whatever other distros that have solved this issue are doing is probably a good start.

**Comment by TheGag96** [comment]
Bumping with a bit more info: In my experience, it seems as though in full screen games, I get really bad screen tearing issues basically all the time. In Pantheon, I never get any screen tearing, but performance can basically switch between a smooth 60 fps, lots of choppiness, back to smooth, then repeat. No idea, but it honestly seems as though the refresh *rate* itself isn't right (like a mismatch between 59.94 FPS or whatever vs. 60). I believe I tried modifying this before just to test it out without luck, so I dunno.

**Comment by TheGag96** [comment]
I noticed that nvidia-450 came out recently and tried it. It appears as though my screen tearing issues are gone! That's great. And when I booted up for the first time with the new drivers, the DE UI (workspace switching, window view) seemed pretty smooth. However, I left my computer on over night, and came back to a really stuttery DE UI. I closed all my programs that were open, and it didn't get smoother. A restart has made it smoother again (for now). It's like something gets out of sync over time... It's hard to say whether it's elementary's fault at all.

**Comment by hugopy** [comment]
I'm also struggling with this! Specially when navigating or randomly while using the system

**Comment by TheGag96** [comment]
It seems an issue on Gala was made a couple years that sounds a lot like this one: https://github.com/elementary/gala/issues/360

**Comment by coppolaemilio** [comment]
Yes, I believe this issue is with Gala, not with the OS itself. It totally seems like the mouse moves at the proper speed but the window manager doesn't.

**Comment by Faaiqishere** [comment]
I also see these weird jitters when opening applications panel or minimizing, maximizing but when closing windows, it closes smoothly... It is to be noted that I dont have a discrete GPU and Im using a intel celeron N3050 CPU and its iGPU with 4 GB of RAM
